fiftyone/factory/repo_factory.py

```diff
@@ -57,15 +57,16 @@ def delegated_operation_repo() -> DelegatedOperationRepo:
     def execution_store_repo(
         dataset_id: Optional[ObjectId] = None,
     ) -> ExecutionStoreRepo:
-        if (
-            MongoExecutionStoreRepo.COLLECTION_NAME
-            not in RepositoryFactory.repos
-        ):
-            RepositoryFactory.repos[
-                MongoExecutionStoreRepo.COLLECTION_NAME
-            ] = MongoExecutionStoreRepo(
+        repo_key = (
```

Comment:
This is a bug fix where dataset_ids were being incorrectly used on "cached" repos.